File: ScaleformUI_Lua/src/Menus/PauseMenus/CrewTag.lua
CrewTag = setmetatable({}, CrewTag)
CrewTag.__index = CrewTag
CrewTag.__call = function()
    return "Tag", "CrewTag"
end

---@class CrewTag
---@field TAG string


---@param tag? string
---@param crewTypeIsPrivate? boolean
---@param crewTagContainsRockstar? boolean
---@param level? number|CrewHierarchy
---@param crewColor? SColor
---@return table
function CrewTag.New(tag, crewTypeIsPrivate, crewTagContainsRockstar, level, crewColor)
    local hexColor
    hexColor = crewColor and crewColor:ToHex() or SColor.HUD_White:ToHex()

    local result = "";
    if tag ~= nil and tag ~= "" then
        if crewTypeIsPrivate then result = result .. "(" else result = result .. " " end
        if crewTagContainsRockstar then result = result .. "*" else result = result .. " " end
        result = result .. level
        result = result .. string.upper(tag)
        result = result .. hexColor
    end
    local data = {
        TAG = result
    }
    return setmetatable(data, CrewTag)
end

File: ScaleformUI_Csharp/Menus/Pause Menus/CrewTag.cs
using CitizenFX.Core;
using ScaleformUI.Elements;
using System.Drawing;

namespace ScaleformUI.PauseMenus
{
    public enum CrewHierarchy
    {
        Leader = 0,
        Commissioner,
        Liutenant = 3, // 2 looks exactly like 1.. so dunno.. let's start on 3
        Representative,
        Muscle,
        Generic
    }

    public class CrewTag
    {
        public string TAG { get; internal set; } = "";
        public CrewTag()
        {
            TAG = "";
        }
        public CrewTag(string tag, bool crewTypeIsPrivate, bool crewTagContainsRockstar, CrewHierarchy level, CitizenFX.Core.Color crewColor)
        {
            try
            {
                string result = "";
                result += crewTypeIsPrivate ? "(" : " ";
                result += crewTagContainsRockstar ? "*" : " ";
                result += ((int)level).ToString();
                result += tag.ToUpper();
                result += $"#{crewColor.R:X2}{crewColor.G:X2}{crewColor.B:X2}";
                TAG = result;
            }
            catch (Exception e)
            {
                Debug.WriteLine(e.ToString());
            }
        }
        public CrewTag(string tag, bool crewTypeIsPrivate, bool crewTagContainsRockstar, CrewHierarchy level, SColor crewColor)
        {
            try
            {
                string result = "";
                result += crewTypeIsPrivate ? "(" : " ";
                result += crewTagContainsRockstar ? "*" : " ";
                result += ((int)level).ToString();
                result += tag.ToUpper();
                result += crewColor.HexValue;
                TAG = result;
            }
            catch (Exception e)
            {
                Debug.WriteLine(e.ToString());
            }
        }
    }
}